examples/packet_forward_test.go

```diff
@@ -0,0 +1,191 @@
+	"github.com/strangelove-ventures/ibctest/testreporter"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zaptest"
+)
+
+func TestPacketForwardMiddleware(t *testing.T) {
+	t.Skip()
+
+	home := t.TempDir()
+	client, network := ibctest.DockerSetup(t)
+
+	rep := testreporter.NewNopReporter()
+	eRep := rep.RelayerExecReporter(t)
+
+	ctx := context.Background()
+
+	cf := ibctest.NewBuiltinChainFactory(zaptest.NewLogger(t), []*ibctest.ChainSpec{
+		{Name: "gaia", ChainName: "gaia-fork", Version: "bugfix-replace_default_transfer_with_router_module"},
+		{Name: "osmosis", ChainName: "osmosis", Version: "v7.3.0"},
+		{Name: "juno", ChainName: "juno", Version: "v6.0.0"},
+	})
+
+	chains, err := cf.Chains(t.Name())
+	require.NoError(t, err)
+
+	gaia, osmosis, juno := chains[0], chains[1], chains[2]
+
+	r := ibctest.NewBuiltinRelayerFactory(ibc.CosmosRly, zaptest.NewLogger(t)).Build(
+		t, client, network, home,
+	)
+
+	const pathOsmoHub = "osmohub"
+	const pathJunoHub = "junohub"
+
+	ic := ibctest.NewInterchain().
+		AddChain(osmosis).
+		AddChain(gaia).
+		AddChain(juno).
+		AddRelayer(r, "relayer").
+		AddLink(ibctest.InterchainLink{
+			Chain1:  osmosis,
+			Chain2:  gaia,
+			Relayer: r,
+			Path:    pathOsmoHub,
+		}).
+		AddLink(ibctest.InterchainLink{
+			Chain1:  gaia,
+			Chain2:  juno,
+			Relayer: r,
+			Path:    pathJunoHub,
+		})
+
+	require.NoError(t, ic.Build(ctx, eRep, ibctest.InterchainBuildOptions{
+		TestName:  t.Name(),
+		HomeDir:   home,
+		Client:    client,
+		NetworkID: network,
+
+		SkipPathCreation: false,
+	}))
+
+	const userFunds = int64(10_000_000_000)
+	users := ibctest.GetAndFundTestUsers(t, ctx, t.Name(), userFunds, osmosis, gaia, juno)
+
+	channels, err := r.GetChannels(ctx, eRep, gaia.Config().ChainID)
+	require.NoError(t, err)
+
+	// Start the relayer on both paths
+	err = r.StartRelayer(ctx, eRep, pathOsmoHub)
+	require.NoError(t, err)
+
+	err = r.StartRelayer(ctx, eRep, pathJunoHub)
+	require.NoError(t, err)
+
+	t.Cleanup(
+		func() {
+			err := r.StopRelayer(ctx, eRep)
+			if err != nil {
+				t.Logf("an error occured while stopping the relayer: %s", err)
+			}
+			for _, c := range chains {
+				if err = c.Cleanup(ctx); err != nil {
+					t.Logf("an error occured while stopping chain %s: %s", c.Config().ChainID, err)
+				}
+			}
+		},
+	)
+
+	// Get original account balances
+	osmosisUser := users[0]
+	gaiaUser := users[1]
+	junoUser := users[2]
+
+	osmosisBalOG, err := osmosis.GetBalance(ctx, osmosisUser.Bech32Address(osmosis.Config().Bech32Prefix), osmosis.Config().Denom)
+	require.NoError(t, err)
+
+	// Send packet from Osmosis->Hub->Juno
+	// receiver format: {intermediate_refund_address}|{foward_port}/{forward_channel}:{final_destination_address}
+	const transferAmount int64 = 100000
+	receiver := fmt.Sprintf("%s|%s/%s:%s", gaiaUser.Bech32Address(gaia.Config().Bech32Prefix), channels[1].PortID, channels[1].ChannelID, junoUser.Bech32Address(juno.Config().Bech32Prefix))
```

Comment:
might be a nice-to-have to expose a function in the module to format these

Reply:
I was kinda thinking the same thing but didn't want to make the `Chain` interface too congested and wasn't sure where a one off utility function like this would live if it was outside of the test.